File: C#/0965. Univalued Binary Tree.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int x) { val = x; }
 * }
 */
public class Solution {
    public bool IsUnivalTree(TreeNode root) {
        if (root==null){
            return true;
        }
        int target=root.val;
        return IsSameValue(root,target);
    }
    
    public bool IsSameValue(TreeNode root,int target){
        if(root==null){
            return true;
        }
        if(root.val!=target){
            return false;
        }
        return (IsSameValue(root.right,target) && IsSameValue(root.left,target));
    }
}

File: archives/20190519python/0965. Univalued Binary Tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def isUnivalTree(self, root: TreeNode) -> bool:
        if not root:
            return True
        return self.sub_isUnivalTree(root,root.val)
    
    def sub_isUnivalTree(self,root,target):
        if not root:
            return True
        return root.val==target and self.sub_isUnivalTree(root.left,target) and self.sub_isUnivalTree(root.right,target)
